scripts/handle-upgrade

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+set -uo pipefail
```

Comment:
As for the business logic of the script... nothing to change, it works remarkably well! Looking forward to flipping to a Python script, but as written, this is just what we need.

Reply:
The reason `-e` is not set is because `qvm-check` returns non-zero if the machine does not exist. We can't assume that it's a run-once-and-forget script (if the first machine doesn't exist, it's possible the others do): If the script is interrupted, or if there's a situation like in #300 where we would consider appending other machines to the script, those commands would never be reached.

Reply:
> The reason -e is not set is because qvm-check returns non-zero if the machine does not exist.

Because all calls to `qvm-check` are in `if` conditionals:

```
$ grep -F qvm-check scripts/handle-upgrade 
	if qvm-check sd-svs --quiet; then
			if qvm-check --running sd-svs; then
	if qvm-check --quiet sd-svs-disp; then
	if qvm-check --quiet sd-export-usb; then
	if qvm-check --quiet sd-gpg; then
		if qvm-check "${template}" --quiet; then
			if qvm-check --running "${template}"; then
```

we needn't worry about the script exiting if one of the machines is missing: the `if` will gobble up any non-zero exit code, causing the conditional to evaluate as false, but `set -e` will _not_ trigger a full exit of the script. For example:

```
[user@dom0 ~]$ cat example.sh
#!/bin/bash
set -e
set -u
set -o pipefail


if qvm-check --quiet non-existent-vm; then
    echo "Yes, we found the VM"
else
    echo "No, that VM does not exist"
fi

echo "Finally, time to exit"
[user@dom0 ~]$ ./example.sh
usage: qvm-check [--verbose] [--quiet] [--help] [--all] [--exclude EXCLUDE]
                 [--running] [--paused] [--template] [--networked]
                 [VMNAME [VMNAME ...]]
qvm-check: error: no such domain: 'non-existent-vm'
No, that VM does not exist
Finally, time to exit
[user@dom0 ~]$ echo $?
0
```

The fact that "Finally, time to exit" is printed means that the script didn't exit early, even with `set -e` in place. Adding `set -e` to the script in this PR will ensure that problems running running e.g. `qvm-shutdown` cause the script to exit—which appears to me to be what we want. We don't want to blithely continue unless each and every one of the steps we list passes, otherwise the state is no longer predictable.

Reply:
Thanks for the clarification @conorsch ! You are correct, just tested locally the script alone as well as the changes in a clean install (which would be the worst case for -e here), and everything works as expected